surface/include/pcl/surface/impl/mls.hpp

```diff
@@ -284,65 +283,21 @@ pcl::MovingLeastSquares<PointInT, PointOutT>::performProcessing (PointCloudOut &
   // Compute the number of coefficients
   nr_coeff_ = (order_ + 1) * (order_ + 2) / 2;
 
-  // Allocate enough space to hold the results of nearest neighbor searches
-  // \note resize is irrelevant for a radiusSearch ().
-  std::vector<int> nn_indices;
-  std::vector<float> nn_sqr_dists;
-  
   size_t mls_result_index = 0;
 
-  // For all points
-  for (size_t cp = 0; cp < indices_->size (); ++cp)
-  {
-    // Get the initial estimates of point positions and their neighborhoods
-    if (!searchForNeighbors ((*indices_)[cp], nn_indices, nn_sqr_dists))
-      continue;
-
-
-    // Check the number of nearest neighbors for normal estimation (and later
-    // for polynomial fit as well)
-    if (nn_indices.size () < 3)
-      continue;
-
-
-    PointCloudOut projected_points;
-    NormalCloud projected_points_normals;
-    // Get a plane approximating the local surface's tangent and project point onto it
-    int index = (*indices_)[cp];
-    
-    if (cache_mls_results_)
-      mls_result_index = index; // otherwise we give it a dummy location.
-
-    computeMLSPointNormal (index, nn_indices, projected_points, projected_points_normals, *corresponding_input_indices_, mls_results_[mls_result_index]);
-
-    // Append projected points to output
-    output.insert (output.end (), projected_points.begin (), projected_points.end ());
-    if (compute_normals_)
-      normals_->insert (normals_->end (), projected_points_normals.begin (), projected_points_normals.end ());
-  }
-
-  // Perform the distinct-cloud or voxel-grid upsampling
-  performUpsampling (output);
-}
-
-//////////////////////////////////////////////////////////////////////////////////////////////
 #ifdef _OPENMP
-template <typename PointInT, typename PointOutT> void
-pcl::MovingLeastSquaresOMP<PointInT, PointOutT>::performProcessing (PointCloudOut &output)
-{
-  // Compute the number of coefficients
-  nr_coeff_ = (order_ + 1) * (order_ + 2) / 2;
-
   // (Maximum) number of threads
-  unsigned int threads = threads_ == 0 ? 1 : threads_;
-
+  const unsigned int threads = threads_ == 0 ? 1 : threads_;
```

Comment:
You mean the added "const"?

I try to "Use const whenever it makes sense.": https://google.github.io/styleguide/cppguide.html#Use_of_const

In this case it's important that the variable does never change, as it defines the size of the vectors for the temporary data.

Reply:
No, `const` is alright. I meant why not using `threads_` directly since OpenMP does not differentiate between 0 and 1. But after a second look I can answer this question myself, this is needed to ensure vectors are properly sized.